### ISSUE TITLE
Unset Prettier options should default to the Prettier defaults

### ISSUE THREAD

**Issue by j0hnm4r5** [xojs/xo#318]
## Problem

I was kind of thrown for a loop earlier: I use Prettier with XO, and left some of the Prettier options unset in my `.prettierrc`, namely the `singleQuote` and `bracketSpacing` options.

This resulted in errors, because [Prettier's default state](https://prettier.io/docs/en/options.html) for those options  is opposite from the [states picked by XO](https://github.com/xojs/xo#prettier) if you leave them unset in your Prettier config.

In general, if you don't edit an option in Prettier, it's because you like its default. And XO should not try to compete with that default (prettier is far more opinionated; it should win out).

## Proposal

If `"prettier": true`, and any of the Prettier config options are unset, use the Prettier defaults, not the ones currently defined in the XO documentation.

**Comment by pvdlg** [comment]
This is an intentional design. The objective of XO is to offer a set of default that works for most and allow to customize them.
XO made several decisions regarding those default before Prettier was added (actually before Prettier was even created) and we decided to keep them, to preserve consistency.

The current behavior might be surprising to you, but doing what you propose would be surprising for everyone else already using XO. The objective here is for XO to use the same default whether you use the `prettier` option or not. So for someone already using XO, enabling the `prettier` flag should just format the code with Prettier but not change the default configuration.

XO made these choices about the default config for several reasons and it was built and refined over time. Prettier (and other projects) made different choices for their default values but that's not a reason to change XO ones.
